### PR TITLE
5839 - Bar-Grouped: Convert Protractor tests to Puppeteer

### LINKED ISSUE #5839: Bar Grouped: Convert Protractor Tests to Puppeteer
**Is your feature request related to a problem or use case? Please describe.**
Convert bar grouped component tests to puppeteer.

**Describe the solution you'd like**
Convert all tests scenarios in protractor to puppeteer. For the naming can be, `{component}.puppeteer-spec.js`

**Describe alternatives you've considered**
N/A

### PR DESCRIPTION
**Pull request by davidcarlsonberg**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Moving to puppeteer from protractor.

**Related github/jira issue (required)**:

Resolves #5839 

**Steps necessary to review your pull request (required)**:

Tests should complete without error in CI.

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by tmcconechy — APPROVED** [review]

**Review by yohannahbautista — DISMISSED** [review]

File: test/components/bar/bar-grouped.puppeteer-spec.js [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 
-    it('should not able to tab through the legends', async () => {
+    it('Should not be able to tab through the legends', async () => {
       // eslint-disable-next-line
         const legendTabIndex = await page.$$eval('.chart-legend-item', e => e.map(el => el.tabIndex));
       expect(legendTabIndex).toEqual([-1, -1, -1, -1]); // These are the values of tabindex of all the legends.
     });
 
-    it('should not select bar on click', async () => {
+    it('Should not select bar on click', async () => {
```
Comment (ericangeles) [inline test/components/bar/bar-grouped.puppeteer-spec.js:20]: We make a pattern where the `it` function will start in lowercase.  E.g., `it('should not select bar on click')`

File: test/components/bar/bar-grouped.puppeteer-spec.js [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+
+  describe('Grouped Bar Chart Disable Selection State Tests', () => {
+    const url = `${baseUrl}/components/bar-grouped/example-disable-selection-state.html`;
 
     beforeAll(async () => {
       await page.goto(url, { waitUntil: ['domcontentloaded', 'networkidle0'] });
     });
 
-    it('should not able to tab through the legends', async () => {
+    it('Should not be able to tab through the legends', async () => {
```
Comment (ericangeles) [inline test/components/bar/bar-grouped.puppeteer-spec.js:14]: It should start in lowercase for all `it` function.

File: test/components/bar/bar-grouped.puppeteer-spec.js [hunk 3]
```diff
@@ -72,4 +75,202 @@ describe('Grouped Bar Chart Puppeteer Tests', () => {
...
+
+  describe('Grouped Bar Chart example-selected tests', () => {
+    const url = `${baseUrl}/components/bar-grouped/test-selected`;
+
+    beforeAll(async () => {
+      await page.goto(url, { waitUntil: ['domcontentloaded', 'networkidle0'] });
+    });
+
+    it('Should not have errors', async () => {
+      await utils.checkForErrors();
```
Comment (ericangeles) [inline test/components/bar/bar-grouped.puppeteer-spec.js:263]: The `checkForErrors` method is working but it needs some adjustments to work in puppeteer properly so I did not include this when converting the protractor to puppeteer.

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Requesting changes to address @ericangeles comments

- lower case
- dont need checkforerrors

**Comment by davidcarlsonberg** [comment]
Looks like a ton of visual regression test issues after pulling in latest `main`. Will this be fixed by #6181 ?

**Comment by tmcconechy** [comment]
Yes trying to... Its a big mess

**Review by tmcconechy — DISMISSED** [review]

File: test/components/bar/bar-grouped.puppeteer-spec.js [hunk 4]
```diff
@@ -72,4 +74,182 @@ describe('Grouped Bar Chart Puppeteer Tests', () => {
...
+  });
+
+  describe('Grouped Bar formatter tests', () => {
+    const url = `${baseUrl}/components/bar-grouped/example-formatter?theme=classic&layout=nofrills`;
+
+    beforeAll(async () => {
+      await page.goto(url, { waitUntil: ['domcontentloaded', 'networkidle0'] });
+    });
+
+    it('Should not visual regress', async () => {
```
Comment (ericangeles) [inline test/components/bar/bar-grouped.puppeteer-spec.js:181]: I think you missed this

File: test/components/bar/bar-grouped.puppeteer-spec.js [hunk 5]
```diff
@@ -72,4 +74,182 @@ describe('Grouped Bar Chart Puppeteer Tests', () => {
...
+  });
+
+  describe('Grouped Bar many groups tests', () => {
+    const url = `${baseUrl}/components/bar-grouped/test-many-groups?theme=classic&layout=nofrills`;
+
+    beforeAll(async () => {
+      await page.goto(url, { waitUntil: ['domcontentloaded', 'networkidle0'] });
+    });
+
+    it('Should not visual regress', async () => {
```
Comment (ericangeles) [inline test/components/bar/bar-grouped.puppeteer-spec.js:206]: Same here

**Review by ericangeles — APPROVED** [review]

**Pull request merged**